sklearn/ensemble/weight_boosting.py

```diff
@@ -109,10 +114,18 @@ def fit(self, X, y, sample_weight=None):
         X, y = check_X_y(X, y, accept_sparse=accept_sparse, dtype=dtype,
                          y_numeric=is_regressor(self))
 
-        if sample_weight is None:
+        if sample_weight is None and transfer_index is None:
             # Initialize weights to 1 / n_samples
             sample_weight = np.empty(X.shape[0], dtype=np.float64)
             sample_weight[:] = 1. / X.shape[0]
+        elif sample_weight is None:
+            # Intialize original sample weights to 1/n_original_samples,
+            # Intialize transfer sample weights to 1/m_tranfer_samples
```

Comment:
Typo in transfer

Reply:
"You will need to document more clearly when and how to use this. I suspect, given the additional fit param and usage, we should consider making this a separate class."

OK, will refactor to TradaBoostClassifier later.

Thanks.